src/tcgwars/logic/impl/gen4/PopSeries8.groovy

```diff
@@ -95,18 +100,23 @@ public enum PopSeries8 implements LogicCardInfo {
             attackRequirement {}
             onAttack {
               damage 20
-              flip { apply PARALYZED }
+              flip { applyAfterDamage(PARALYZED) }
             }
           }
           move "Fire Spin", {
             text "90 damage. Discard 2 Basic Energy cards attached to Heatran. (If you can’t discard cards, this attack does nothing."
             energyCost R, R, R, C
-            attackRequirement {}
+            attackRequirement {
+              assert self.cards.filterByType(BASIC_ENERGY).size() >= 2 : "Needs at least 2 Basic Energy cards attached"
+            }
             onAttack {
-              damage 0
+              damage 90
+
+              afterDamage {
+                self.cards.filterByType(BASIC_ENERGY).select(count: 2, "Discard 2 Basic energy cards from $self.").discard()
```

Comment:
The cards that are going to be discarded should be selected before damage.

Reply:
To be clear, select before damage, discard afterDamage.